hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -300,12 +301,11 @@ protected OmMetadataManagerImpl() {
     return deletedDirTable;
   }
 
-  @Override
-  public Table<String, OmKeyInfo> getOpenKeyTable() {
-    if (OzoneManagerRatisUtils.isBucketFSOptimized()) {
-      return openFileTable;
+  public Table getOpenKeyTable(BucketLayout bucketLayout) {
+    if (bucketLayout.equals(BucketLayout.OBJECT_STORE)) {
```

Comment:
@aryangupta1998 thanks for the patch. Like we discussed offline to fix the test failure, could you modify the condition to include the LEGACY case.

   ```
 if (bucketLayout.equals(BucketLayout.FILE_SYSTEM_OPTIMIZED)) {
      return openFileTable;
    }
    return openKeyTable;
```

Reply:
Done